src/python/grpcio/grpc/_cython/_cygrpc/call.pyx.pxi

```diff
@@ -106,6 +107,8 @@ cdef class Call:
   def __dealloc__(self):
     if self.c_call != NULL:
       grpc_call_destroy(self.c_call)
+    grpc_shutdown()
```

Comment:
Just because we cleared out our surface API handles doesn't necessarily mean that the underlying core has cleared out its own handles (it does its own internal reference counting). It seems plausible to me that, given gRPC doesn't set up its own daemon, unclean shutdown won't be blocked on and unordered events may occur (module unloading, Python object deallocation, and at-exit handling). With or without parity, if events occur out of order, it seems reasonable to me that things may leak.

`__dealloc__` functions in Cython are directly attached to the CPython `tp_dealloc` field; short of a (strangely-specific-to-us) bug in Cython or a major (also strangely-specific-to-us) bug in PyPy's CPython handling I'd imagine that such functions would always be called on clean shutdowns.

I guess now I'm mildly curious about what happens if `grpc_shutdown` isn't called at the base of this PR...